code/modules/mob/living/simple_animal/hostile/ordeal/misc.dm

```diff
@@ -0,0 +1,109 @@
+	melee_damage_lower = 35
+	melee_damage_upper = 35
+	ranged = TRUE
+	attack_verb_continuous = "bashes"
+	attack_verb_simple = "bashes"
+	damage_coeff = list(BRUTE = 1, RED_DAMAGE = 0.7, WHITE_DAMAGE = 0.4, BLACK_DAMAGE = 0.0, PALE_DAMAGE = 0.4)
+	move_resist = MOVE_FORCE_OVERPOWERING
+	projectiletype = /obj/projectile/black
+	var/pulse_cooldown
+	var/pulse_cooldown_time = 15 SECONDS
+	var/pulse_damage = 30
+	var/busy = FALSE
+
+/mob/living/simple_animal/hostile/ordeal/black_dusk/Life()
+	. = ..()
+	if(!.) // Dead
+		return FALSE
+	if(pulse_cooldown < world.time)
+		pulse()
+
+/obj/projectile/black
+	name = "kunai"
+	icon_state = "blackfixer"
+	knockdown = 5
+	damage = 30
+	damage_type = BLACK_DAMAGE
+
+/mob/living/simple_animal/hostile/ordeal/black_dusk/proc/pulse()
+	icon_state = "fixer_b_attack"
+	SLEEP_CHECK_DEATH(5)
+	pulse_cooldown = world.time + pulse_cooldown_time
+	playsound(src, 'sound/weapons/resonator_blast.ogg', 100, FALSE, 90)
+	for(var/mob/living/L in urange(40, src))
+		if(faction_check_mob(L))
+			continue
+		L.apply_damage(pulse_damage, BLACK_DAMAGE, null, L.run_armor_check(null, BLACK_DAMAGE), spread_damage = TRUE)
+	for(var/obj/machinery/computer/abnormality/A in urange(16, src))
+		if(prob(66) && !A.meltdown && A.datum_reference && A.datum_reference.current && A.datum_reference.qliphoth_meter)
+			A.datum_reference.qliphoth_change(pick(-1, -2))
+	SLEEP_CHECK_DEATH(5)
+	icon_state = "fixer_b"
+
+/mob/living/simple_animal/hostile/ordeal/black_dusk/AttackingTarget()
+	//If he doesn't attack for about 15 seconds, pulse.
+	pulse_cooldown = world.time + pulse_cooldown_time
+	if(busy)
+		return
+	..()
+	if(prob(30))
+		combo()
+
+/mob/living/simple_animal/hostile/ordeal/black_dusk/proc/combo()
+	busy = TRUE
+	for(var/i = 1 to 6)
+		i++
+		for(var/mob/living/carbon/human/L in range(1, src))
+			var/turf/open/T = get_turf(L)
+			new /obj/effect/temp_visual/small_smoke/halfsecond(T)
+			playsound(get_turf(src), 'sound/weapons/ego/hammer.ogg', 75, 1)
+			L.apply_damage(10, BLACK_DAMAGE, null, L.run_armor_check(null, BLACK_DAMAGE), spread_damage = TRUE)
+			SLEEP_CHECK_DEATH(3)
+	SLEEP_CHECK_DEATH(10)
+	busy = FALSE
+
+//Yeah so this midnight is supposed to be weak as shit.
+/mob/living/simple_animal/hostile/ordeal/pink_midnight
+	name = "A Party Everlasting"
+	desc = "An overturned teacup, a party everlasting."
+	icon = 'ModularTegustation/Teguicons/64x96.dmi'
+	icon_state = "party"
+	icon_living = "party"
+	faction = list("pink_midnight")
+	pixel_x = -16
+	base_pixel_x = -16
+	maxHealth = 2000
+	health = 2000
+	melee_damage_type = PALE_DAMAGE
+	armortype = PALE_DAMAGE
+	rapid_melee = 2
+	melee_damage_lower = 14
+	melee_damage_upper = 14
+	attack_verb_continuous = "bashes"
+	attack_verb_simple = "bashes"
+	damage_coeff = list(BRUTE = 1, RED_DAMAGE = 1, WHITE_DAMAGE = 1, BLACK_DAMAGE = 1, PALE_DAMAGE = 1)
+
+//Breach everything, Drag to her, for one big jambouree
+/mob/living/simple_animal/hostile/ordeal/pink_midnight/Initialize()
```

Comment:
Imagine what happens if admins want to spawn it alone. Move this to the ordeal datum instead.

Reply:
The... Whole point of it is that it drags everything to it?

Reply:
I tried it during a test, need to recode it.
That's kinda this whole monster's thing though.